### DISCUSSION TITLE
Textsystem

### DISCUSSION THREAD

**Discussion by rgwch** [elexis/elexis.github.io#4] [General]
While Elexis 3.12 now runs smoothly with my Linux- and Windows- Systems on Java 17, I'm still stuck at a point:

What is the recommended way to update with respect to the text system? There is no OpenOffice plugin anymore, and the new docx systems seems to accept docx files only. 
Manually converting seven zillions of existing documents is obviously not the correct way to deal with this.

I found a "Dokumente Konvertierung" plugin, which looks like something for that purpose, but entering the default REST URL https://office.medelexis.ch/jodconverter does not seem do anything.

Any hints?

**Comment by debenol1** [comment]
Up to now we are using the Hilotec/LibreOffice combination. Due to exhausting storage issues we intend to replace the Hilotec Textplugin with the Docx Textplugin which can be used with LibreOffice. The only drop of bitterness is the fact that we have to switch to the docx file format...

**Comment by col-panic** [comment]
As far as I know (please @huthomas @kappisj correct me), the "new" default is the [Docx Generator](https://github.com/elexis/elexis-3-core/tree/master/bundles/at.medevit.elexis.text.docx) which generates docx documents out of docx templates with variables inside. For [ElexisEnvironment](https://github.com/elexis/elexis-environment) we are working on subsequent editing capability with [onlyoffice](https://github.com/ONLYOFFICE) after this generation.

The JodConverter has another purpose, resp. is used for another purpose - it converts documents to pdf before sending them via mail.
It is not meant to be used via the webbrowser.

**Comment by kappisj** [comment]
The recommended way is, indeed, to use the docx-plugin. It gives you the ability to choose the editor of your choice. 

For the conversion from a LibreOffice to the new docx plugin you have to convert at least the forms (Vorlagen) manually which can mostly be done by opening the exisiting .odt and save it as .docx. Medelexis also provides a set of most-used-.docx-forms which you can find [here](https://support.medelexis.ch/docs/briefvorlagen/)

After this we usually set that .docx is openend with word and .odt is openend with LibreOffice on OS-level.
